server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -761,6 +774,17 @@ protected boolean cleanupAccount(AccountVO account, long callerUserId, Account c
                     s_logger.error("Unable to expunge vm: " + vm.getId());
                     accountCleanupNeeded = true;
                 }
+                else if (!vm.getState().equals(VirtualMachine.State.Destroyed)) {
+                    // We have to emit the event here because the state listener is ignoring root volume deletions,
+                    // assuming that the UserVMManager is responsible for emitting the usage event for them when
```

Comment:
Why not fix the state listener, or the manager that is responsible for emitting the event in the first place?